src/attack.rs

```diff
@@ -494,9 +490,9 @@ fn boss_attack(
                         .insert(Attack { damage: 30 })
                         .insert(AttackFrames {
                             //TODO: Check if this is correct
-                            startup: 1,
-                            active: 2,
-                            recovery: 3,
+                            startup: 6,
+                            active: 4,
+                            recovery: 5,
```

Comment:
startup, active and recovery are all measured from the beginning of the animation.
I was thinking about if there was a better way to represent this than ints, like ranges of frames, to better communicate intent here.

The startup frame listed there is the last frame of startup, which starts at frame 0 of the animation and goes to `startup`.
The active frame is the last active frame, also measured from frame 0 of the animation.

(recovery is measured from beginning as well, right now its a bit redundant but the idea is that when we have chaining system in place it will be important to determine cancel windows for moves. It should not be > than the total frames of the animation, and conceptually represents the frames from when the hitbox deactivates, to the frame when you are able to cancel an attack/animation with a new input.) For now you can think of recovery as the last frame off the animation.

Right now as these are represented, their should not be a situation where startup > active, or active > recovery.

Reply:
So it's the frame index and not the number of frames? So 5 - 9 and 14 respectively.

Reply:
Yup they are indexes